### PR TITLE
Add tests publish serialized bad arguments

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: test_rmw_implementation/test/test_publisher.cpp [hunk 1]
```diff
@@ -375,3 +375,32 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
...
+
+  const char * implementation_identifier = pub->implementation_identifier;
+  pub->implementation_identifier = "not-an-rmw-implementation-identifier";
+  ret = rmw_publish_serialized_message(pub, &serialized_message, default_allocation);
+  EXPECT_EQ(RMW_RET_INCORRECT_RMW_IMPLEMENTATION, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+  pub->implementation_identifier = implementation_identifier;
+
+  EXPECT_EQ(RMW_RET_OK, rmw_serialized_message_fini(&serialized_message)) <<
+    rmw_get_error_string().str;
```
Comment (Blast545) [inline test_rmw_implementation/test/test_publisher.cpp:405]: Do this passes uncrustify as it is now?
```suggestion
  EXPECT_EQ(
     RMW_RET_OK, rmw_serialized_message_fini(&serialized_message)) << rmw_get_error_string().str;
```

**Review by Blast545 — APPROVED** [review]
Minor comment, LGTM after green CI.

As for the name of the PR, is somewhat long, maybe something like "Add tests publish serialized" should do. You can use [These guidelines](https://chris.beams.io/posts/git-commit/) to have an idea of the expectations for PR titles

File: test_rmw_implementation/test/test_publisher.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp:405]: Yes lol, but your suggestion is better

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp]: *updated

File: test_rmw_implementation/test/test_publisher.cpp [hunk 2]
```diff
@@ -375,3 +375,32 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+TEST_F(
+  CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION),
+  publish_serialized_message_with_bad_arguments) {
+  rmw_publisher_allocation_t * default_allocation{nullptr};
```
Comment (hidmic) [inline test_rmw_implementation/test/test_publisher.cpp:382]: @Lobotuerk nit: 
```suggestion
  rmw_publisher_allocation_t * null_allocation{nullptr};  // also valid
```

**Review by hidmic — APPROVED** [review]

File: test_rmw_implementation/test/test_publisher.cpp (hunk 2 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_publisher.cpp]: updated

**Comment by hidmic** [comment]
@ros-pull-request-builder retest this please.

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12395/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7363/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10106/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12308/)

**Comment by Lobotuerk** [comment]
CI up to test_rmw_implementation after rebase:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12430/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7397/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10140/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12343/)

**Comment by Lobotuerk** [comment]
Merging with green CI

**Pull request merged**